src/textual/css/stylesheet.py

```diff
@@ -636,18 +636,28 @@ def update_nodes(self, nodes: Iterable[DOMNode], animate: bool = False) -> None:
         rules_map = self.rules_map
         apply = self.apply
 
-        for node in nodes:
-            rules = {
+        def get_rules(node: DOMNode) -> set[RuleSet]:
+            """Get set of rules for the given node."""
+            return {
                 rule
-                for name in node._selector_names
-                if name in rules_map
+                for name in rules_map.keys() & node._selector_names
                 for rule in rules_map[name]
             }
-            apply(node, limit_rules=rules, animate=animate)
+
+        for node in nodes:
+            rules = get_rules(node)
+            if rules:
```

Comment:
Possibly silly question:

If `rules` is an empty set, would this prevent the update from working in the case where we go from some rules to no rules on the node? Or can rules not be empty?

Reply:
If `rules` is empty, then it means that none of the selectors could possibly match the node. i.e something completely unstyled.